samples/server/aspnet/Program.cs

```diff
@@ -1,25 +1,139 @@
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Threading.Tasks;
-using Microsoft.AspNetCore.Hosting;
-using Microsoft.Extensions.Configuration;
-using Microsoft.Extensions.Hosting;
-using Microsoft.Extensions.Logging;
-
-namespace GallagherUniversityStudentPortalSampleSite
+Database.Setup(resetDatabase: false).Wait();
+
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+builder.Services.AddControllersWithViews().AddNewtonsoftJson();
+
+builder.Services.AddTransient<Database>();
+
+var apiClientFactory = new GglApiClientFactory(
+    builder.Configuration["host"],
+    builder.Configuration["apiKey"],
+    builder.Configuration["clientCertificatePfx"],
+    builder.Configuration["clientCertificatePfxPassword"]);
+
+builder.Services.AddSingleton(apiClientFactory);
+
+Configuration = builder.Configuration;
+
+var commandCentreConfig = InitialiseCommandCentreConfiguration(apiClientFactory.CreateClient(), Configuration).Result;
+
+builder.Services.AddSingleton(commandCentreConfig);
+
+builder.Services.AddTransient(_ => apiClientFactory.CreateClient());
+
+var app = builder.Build();
+
+// Configure the HTTP request pipeline.
+if (!app.Environment.IsDevelopment())
+{
+    app.UseExceptionHandler("/Home/Error");
+    app.UseHsts();
+}
+
+app.UseHttpsRedirection();
+app.UseStaticFiles();
+
+app.UseRouting();
+
+app.UseAuthorization();
+
+app.MapControllerRoute(
+    name: "default",
+    pattern: "{controller=Home}/{action=Index}/{id?}");
+
+app.Run();
+
+public partial class Program
```

Comment:
Since we're now using the .NET 6 framework, might as well use the prescribed minimal hosting model for ASP.NET Core 6 applications. You get less lines of code and is cleaner.